File: solutions/clojure/resistor-color-duo/1/src/resistor_color_duo.clj
(ns resistor-color-duo)

(def colors-map
  { "black" 0
    "brown" 1
    "red" 2
    "orange" 3
    "yellow" 4
    "green" 5
    "blue" 6
    "violet" 7
    "grey" 8
    "white" 9 })

(defn resistor-value
  "Returns the resistor value based on the given colors"
  [colors]
  (->> colors
       (take 2)
       (map colors-map)
       (apply str)
       parse-long))

File: solutions/zig/resistor-color-duo/1/resistor_color_duo.zig
const std = @import("std");

pub const ColorBand = enum {
    black,
    brown,
    red,
    orange,
    yellow,
    green,
    blue,
    violet,
    grey,
    white,
};

pub fn colorCode(colors: [2]ColorBand) usize {
    var sum: usize = 0;
    for (colors) |color| {
        sum = sum * 10 + @intFromEnum(color);
    }
    return sum;
}
